heroic-core/src/main/java/com/spotify/heroic/CoreQueryManager.java

```diff
@@ -97,12 +97,16 @@
     private final QueryCache queryCache;
     private final AggregationFactory aggregations;
     private final OptionalLimit groupLimit;
+    private boolean logQueries;
+    private OptionalLimit logQueriesThresholdDataPoints;
```

Comment:
Touched on this subject earlier today with @juruen. He commented that he had seen that when a big query hit the system it would also slow down other queries, so we'd perhaps see a large swath of queries being logged at that time. I'll add time based threshold tho.